packages/k2-alpine/src/components/Button/Button.tsx

```diff
@@ -0,0 +1,231 @@
+          ]}>
+          <WrapperComponent
+            style={{
+              alignItems: 'center',
+              marginHorizontal: 8,
+              justifyContent: 'center',
+              width: '100%',
+              backgroundColor
+            }}
+            {...(shouldUseBlurWrapper && {
+              glassType: theme.isDark ? 'dark3' : 'light2'
+            })}>
+            <View
+              style={{
+                flexDirection: 'row',
+                alignItems: 'center',
+                ...sizeStyles[size]
+              }}>
+              {leftIcon ? (
+                getIcon(leftIcon, {
+                  width: iconWidth,
+                  height: iconWidth,
+                  color: tintColor,
+                  style: { marginRight: 8 }
+                })
+              ) : rightIcon ? (
+                <View style={{ width: iconWidth, marginRight: 8 }} />
+              ) : null}
+              <Text
+                numberOfLines={1}
+                // todo: please remove the ts-ignore comments after we address this issue
+                // https://ava-labs.atlassian.net/browse/CP-9297
+                // @ts-ignore
+                variant={textVariant}
+                adjustsFontSizeToFit={Platform.OS === 'ios'}
+                style={{
+                  color: tintColor,
+                  flexShrink: 1
+                }}>
+                {children}
+              </Text>
+              {rightIcon ? (
+                getIcon(rightIcon, {
+                  width: iconWidth,
+                  height: iconWidth,
+                  color: tintColor,
+                  style: { marginLeft: 8 }
+                })
+              ) : leftIcon ? (
+                <View style={{ width: iconWidth, marginLeft: 8 }} />
+              ) : null}
+            </View>
+          </WrapperComponent>
+        </View>
+      </TouchableOpacity>
+    )
+  }
+)
+
+const sizeStyles = StyleSheet.create({
+  large: {
+    paddingHorizontal: 12,
+    minHeight: 42
+  },
+  medium: {
+    paddingHorizontal: 12,
+    minHeight: 36
+  },
+  small: {
+    paddingHorizontal: 8,
+    minHeight: 27
+  }
+})
+
+const iconComponents = {
+  check: Icons.Navigation.Check,
+  expandMore: Icons.Navigation.ExpandMore
+}
+
+const getIcon = (
+  type: ButtonIconType,
+  iconProps: {
+    width: number
+    height: number
+    color: string
+    style?: ViewStyle
+  }
+): JSX.Element | undefined => {
+  const IconComponent = iconComponents[type]
+  return <IconComponent {...iconProps} />
+}
+
+const getBackgroundColor = (
+  type: ButtonType,
+  theme: K2AlpineTheme,
+  disabled: boolean | undefined
+): string | undefined => {
+  if (disabled) {
+    return theme.isDark
+      ? overlayColor(alpha(colors.$neutralWhite, 0.3), colors.$neutral850)
+      : overlayColor(alpha(colors.$neutral850, 0.3), colors.$neutralWhite)
+  }
+
+  switch (type) {
+    case 'primary':
+      return theme.isDark ? colors.$neutralWhite : colors.$neutral850
```

Comment:
ideally we shouldn't need to do this and should just use surfacePrimary. if it is like this in the design then we should ping Todd/Michael